common/lib/xmodule/xmodule/js/src/video/09_video_caption.js

```diff
@@ -1137,6 +1139,8 @@
                     .removeClass('is-active')
                     .find('.control-text')
                         .text(gettext('Turn on closed captioning'));
+
```

Comment:
That's a good question. I don't know if there are patterns, but my inclination would be to send the event _after_ the function completed. No point in firing an event if whatever was supposed to happen never happens. If you agree with that, then I should move these triggers after the function.